vm_supervisor/resources.py

```diff
@@ -0,0 +1,132 @@
+from datetime import datetime, timezone
+from functools import lru_cache
+from typing import Tuple
+
+import cpuinfo
+import psutil
+from aiohttp import web
+from aleph_message.models.program import CpuProperties
+from pydantic import BaseModel
+
+from .conf import settings
+
+
+class Period(BaseModel):
+    datetime: datetime
+
+
+class LoadAverage(BaseModel):
+    load1: float
+    load5: float
+    load15: float
+
+    @classmethod
+    def from_psutil(cls, psutil_loadavg: Tuple[float, float, float]):
+        return cls(load1=psutil_loadavg[0],
+                   load5=psutil_loadavg[1],
+                   load15=psutil_loadavg[2],
+                   )
+
+
+class CoreFrequencies(BaseModel):
+    min: float
+    max: float
+
+    @classmethod
+    def from_psutil(cls, psutil_freq: psutil._common.scpufreq):
+        min = psutil_freq.min or psutil_freq.current
+        max = psutil_freq.max or psutil_freq.current
+        return cls(min=min, max=max)
+
+
+class CpuUsage(BaseModel):
+    count: int
+    load_average: LoadAverage
+    core_frequencies: CoreFrequencies
+
+
+class MemoryUsage(BaseModel):
+    total_kB: int
```

Comment:
Nitpick: not a fan of the case choice here, maybe `total_kb`?

Reply:
Uppercase is the symbol for `byte`, lowercase for `bit`. These should not be confused.

Reply:
True that.